app/services/analytics.rb

```diff
@@ -42,13 +42,22 @@ def track_mfa_submit_event(attributes)
   attr_reader :user, :request, :sp, :ahoy
 
   def request_attributes
-    {
+    attributes = {
       user_ip: request.remote_ip,
       hostname: request.host,
       pid: Process.pid,
       service_provider: sp,
       trace_id: request.headers['X-Amzn-Trace-Id'],
-    }.merge!(browser_attributes)
+    }
+
+    attributes[:git_sha] = IdentityConfig::GIT_SHA
+    if IdentityConfig::GIT_TAG.present?
+      attributes[:git_tag] = IdentityConfig::GIT_TAG
+    else
+      attributes[:git_branch] = IdentityConfig::GIT_BRANCH
+    end
```

Comment:
if we do this conditional logic... WDYT of throwing on a `stub_const` in a spec helper for these configs? that way way it doesn't matter if CI is run against a branch or a tag?

Reply:
ah hadn't considered that. added stub_const to the specs to avoid that problem in 0e96ac2c7

Reply:
I'd stub `GIT_TAG` and `GIT_BRANCH` too but ya perfect thx

Reply:
I am a stubbing minimalist 😬